### PR TITLE
Handle TSV/CSV data for program/project creation + debug logs

### PR DESCRIPTION
**Pull request by paulineribeyre**
### Improvements
- Add debug log when unable to parse JSON input
- Handle TSV/CSV data for program/project creation

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
The style in this PR agrees with [`black`](https://github.com/ambv/black). :heavy_check_mark:

This formatting comment was generated automatically by a script in [uc-cdis/wool](https://github.com/uc-cdis/wool).

File: sheepdog/blueprint/routes/views/__init__.py [hunk 1]
```diff
@@ -112,13 +112,22 @@ def root_create():
...
-    node_id = None
-    doc = parse.parse_request_json()
+    input_doc = flask.request.get_data().decode("utf-8")
+    content_type = flask.request.headers.get("Content-Type", "").lower()
+    if content_type == "text/csv":
+        doc, _ = utils.transforms.CSVToJSONConverter().convert(input_doc)
+    elif content_type in ["text/tab-separated-values", "text/tsv"]:
+        doc, _ = utils.transforms.TSVToJSONConverter().convert(input_doc)
+    else:
+        doc = utils.parse.parse_request_json()
```
Comment (johnfrancismccann) [inline sheepdog/blueprint/routes/views/__init__.py:122]: If possible, a minor suggestion would be to check if any errors are returned from the `convert` calls.

File: sheepdog/blueprint/routes/views/program/__init__.py [hunk 2]
```diff
@@ -137,12 +137,22 @@ def create_project(program):
...
-    res = None
-    doc = utils.parse.parse_request_json()
+    input_doc = flask.request.get_data().decode("utf-8")
+    content_type = flask.request.headers.get("Content-Type", "").lower()
+    if content_type == "text/csv":
+        doc, _ = utils.transforms.CSVToJSONConverter().convert(input_doc)
+    elif content_type in ["text/tab-separated-values", "text/tsv"]:
+        doc, _ = utils.transforms.TSVToJSONConverter().convert(input_doc)
+    else:
+        doc = utils.parse.parse_request_json()
```
Comment (johnfrancismccann) [inline sheepdog/blueprint/routes/views/program/__init__.py:147]: Same as above.

**Review by johnfrancismccann — APPROVED** [review]
Left a small suggestion, but LGTM.

**Pull request merged**